docs/development/javascript/how-to-build-and-use-vuejs-components/index.md

```diff
@@ -2,388 +2,561 @@
+<rating-counter v-bind:parent="parentHeader">
+    <h1>{{ parentHeader.label }}</h1>
+</rating-counter>
 ```
 
-Slot nesting: This is probably the most important feature of slots the ability to use components within components. That is especially useful when creating structure for your apps! Lets modify our example again:
-```html
+Using the `parentHeader.label` string here wouldn't make much sense anyway, so this would only serve to demonstrate the scope concept.
+{{< /disclosure-note >}}
+
+### Nesting Slots
+
+The most important feature of slots might be the ability to use components within components. This is especially useful when creating structure for your apps.
+
+Replace the contents of your `ratingcounter.html` with the following snippet:
+
+{{< file "ratingcounter.html" html >}}
 <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
 
 <div id="app">
-   <rating-counter v-bind:parent="parentTitle"><rating-title></rating-title></rating-counter>
-   <rating-counter v-bind:parent="parentTitle"><h2>Rating 2</h2></rating-counter>
-   <rating-counter v-bind:parent="parentTitle"><h3>Rating 3</h3></rating-counter>
+    {{ parentHeader.label }}{{ parentHeader.globalCount }}
+    <hr />
+
+    <rating-counter v-bind:parent="parentHeader">
+        <rating-title>Rating 1</rating-title>
+    </rating-counter>
+
+    <rating-counter v-bind:parent="parentHeader">
+        <rating-title>Rating 2</rating-title>
+    </rating-counter>
+
+    <rating-counter v-bind:parent="parentHeader">
+        <rating-title></rating-title>
+    </rating-counter>
 </div>
+
 <script>
-   Vue.component('rating-counter', {
-       props: ['parent'],
-       data() {
-           return {
-               count: 0
-           }
-       },
-       template:   `<div>
-                       <slot></slot>
-                       <button v-on:click="count--;parent.counter--;">Thumbs Down</button>
-                       {{ count }}
-                       <button v-on:click="count++;parent.counter++;">Thumbs up</button>
-                   </div>`
-   })
-
-   Vue.component('rating-title', {
-       template:   `<div>
-                       <slot><h1>Hello rating title</h1></slot>
-                   </div>`
-   })
-    new Vue({
-       el: '#app',
-       data: {
-           parentTitle: {
-               title: "Counter is at: ",
-               counter: 0
-           }
-       }
-   })
+Vue.component('rating-counter', {
+    props: ['parent'],
+    data() {
+        return {
+            count: 0
+        }
+    },
+    template:   `<div>
+                    <slot></slot>
+                    <button v-on:click="count--;parent.globalCount--;">Thumbs Down</button>
+                    {{ count }}
+                    <button v-on:click="count++;parent.globalCount++;">Thumbs Up</button>
+                </div>`
+})
+
+Vue.component('rating-title', {
+    template:   `<div>
+                    <h1>
```

Comment:
Moved the slot reference inside the rating-title component template to inside the h1 tags. This allows someone to just pass a string inside the rating-title tags